crates/node/src/command.rs

```diff
@@ -52,69 +52,138 @@ impl SubstrateCli for Cli {
+fn set_chain_spec(cli: &mut Cli) -> Result<(), String> {
+    // checks if it should retrieve and enable a specific chain-spec
+    let madara_path = cli.run.madara_path.clone().expect("Failed retrieving madara_path").to_str().unwrap().to_string();
+    let local_path = utils::get_project_path();
+
+    match (
+        cli.run.chain_spec_url.clone(),
+        cli.run.testnet,
+        cli.run.disable_url_fetch,
+        cli.run.disable_madara_configs,
+        local_path,
+    ) {
+        (Some(chain_spec_url), _, _, _, _) => {
+            utils::fetch_from_url(
+                chain_spec_url.clone(),
+                madara_path.clone() + "/chain-specs",
+                cli.run.update_configs,
+            )?;
+            let chain_spec = chain_spec_url.split('/').last().expect("Chain spec file name not found");
+            cli.run.run_cmd.shared_params.chain = Some(madara_path + "/chain-specs/" + chain_spec);
+        }
+        (None, Some(Testnet::Sharingan), false, false, Ok(ref src_path)) => {
+            let src_path = src_path.clone() + "/configs/chain-specs/testnet-sharingan-raw.json";
+            utils::copy_from_filesystem(src_path, madara_path.clone() + "/chain-specs", cli.run.update_configs)?;
+            cli.run.run_cmd.shared_params.chain = Some(madara_path + "/chain-specs/testnet-sharingan-raw.json");
+        }
+        (None, Some(Testnet::Sharingan), false, false, _) => {
+            let madara_configs: configs::Configs =
+                serde_json::from_str(&utils::read_file_to_string(madara_path.clone() + "/configs/index.json")?)
+                    .expect("Failed to load madara configs");
+            // If testnet flag is specified, right now it's only Sharingan
+            for chain_spec in madara_configs.chain_specs {
+                configs::fetch_and_validate_file(
+                    madara_configs.remote_base_path.clone(),
+                    chain_spec,
+                    madara_path.clone() + "/configs/chain-specs/",
+                    cli.run.update_configs,
                 )?;
             }
+            cli.run.run_cmd.shared_params.chain = Some(madara_path + "/chain-specs/testnet-sharingan-raw.json");
         }
+        _ => {}
     }
 
-    // TODO confirm with the CI that we are fetching all and fetch dynamically
-    // Issue #982
-    for file in constants::CAIRO_CONTRACTS_FILES {
-        let src_path = utils::get_project_path();
-        if let Ok(src_path) = src_path {
-            let src_path = src_path + "/configs/cairo-contracts/" + file;
-            utils::copy_from_filesystem(src_path, madara_path.clone() + "/cairo-contracts")?;
-        } else {
-            utils::fetch_from_url(
-                constants::CAIRO_CONTRACTS_URL.to_string() + file,
-                madara_path.clone() + "/cairo-contracts",
+    if cli.run.run_cmd.shared_params.chain.is_some() {
+        cli.run.run_cmd.rpc_external = true;
+        cli.run.run_cmd.rpc_methods = RpcMethods::Unsafe;
+    }
```

Comment:
I need a comment to understand why are we doing this

Reply:
We had that since the --testnet flag was created. The reasoning was for authority node operators to run their nodes with a command that we had set, between iterations. So they would always run `madara --testnet sharingan`, and if we needed to enable some extra stuff, we could without an issue. We can remove that, if u want

Reply:
It sound usefull. We can keep it, but we have to document it heavily and flag it as TO: remove when going in production